### PR TITLE
Better error reporting for built-in TDS compilation

### PR DESCRIPTION
**Pull request by bwaresiak**
Task/Issue URL: https://app.asana.com/0/856498667320406/1200617159785081
Tech Design URL:
CC:

**Description**:

Error information in case compilation of Local Tracker Data set fails - this is super rare case.

**Steps to test this PR**:

General smoke tests around TDS recompilation/update.

**Device Testing**:

* [ ] iPhone
* [ ] iPad

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: Core/Pixel.swift [hunk 1]
```diff
@@ -212,10 +212,11 @@ public struct PixelParameters {
     static let dataAvailiability = "dp"
     
     static let errorCode = "e"
-    static let errorDesc = "d"
+    static let errorDomain = "d"
+    static let errorDescription = "de"
     static let errorCount = "c"
     static let underlyingErrorCode = "ue"
-    static let underlyingErrorDesc = "ud"
+    static let underlyingErrorDomain = "ud"
```
Comment (samsymons) [inline Core/Pixel.swift:219]: :+1:

File: Core/Pixel.swift [hunk 2]
```diff
@@ -285,20 +286,23 @@ public class Pixel {
 
 extension Pixel {
     
-    public static func fire(pixel: PixelName, error: Error, withAdditionalParameters params: [String: String] = [:]) {
+    public static func fire(pixel: PixelName,
+                            error: Error,
+                            withAdditionalParameters params: [String: String] = [:],
+                            onComplete: @escaping (Error?) -> Void = {_ in }) {
```
Comment (samsymons) [inline Core/Pixel.swift]: Tiny whitespace note here, I don't really mind this one but wonder if SwiftLint will complain. If it doesn't, then this is fine to leave 🙂 

```suggestion
                            onComplete: @escaping (Error?) -> Void = { _ in }) {
```

File: Core/Pixel.swift [hunk 3]
```diff
@@ -285,20 +286,23 @@ public class Pixel {
...
             newParams[PixelParameters.underlyingErrorCode] = "\(underlyingError.code)"
-            newParams[PixelParameters.underlyingErrorDesc] = underlyingError.domain
+            newParams[PixelParameters.underlyingErrorDomain] = underlyingError.domain
         } else if let sqlErrorCode = nsError.userInfo["NSSQLiteErrorDomain"] as? NSNumber {
             newParams[PixelParameters.underlyingErrorCode] = "\(sqlErrorCode.intValue)"
-            newParams[PixelParameters.underlyingErrorDesc] = "NSSQLiteErrorDomain"
+            newParams[PixelParameters.underlyingErrorDomain] = "NSSQLiteErrorDomain"
         }
-        fire(pixel: pixel, withAdditionalParameters: newParams)
+        fire(pixel: pixel, withAdditionalParameters: newParams, includeATB: false, onComplete: onComplete)
```
Comment (samsymons) [inline Core/Pixel.swift:305]: Cool, glad we're omitting ATB for these going forward.

**Review by samsymons — APPROVED** [review]

**Comment by bwaresiak** [comment]
Thanks @samsymons !

**Pull request merged**